x-pack/plugins/security_solution/public/timelines/components/side_panel/new_user_detail/hooks.ts

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo, useEffect } from 'react';
+import * as i18n from './translations';
+import type { ManagedUserTable, ObservedUserData, ObservedUserTable } from './types';
+import type { AzureManagedUser } from '../../../../../common/search_strategy/security_solution/users/managed_details';
+
+import {
+  Direction,
+  UsersQueries,
+  NOT_EVENT_KIND_ASSET_FILTER,
+} from '../../../../../common/search_strategy';
+import { useSourcererDataView } from '../../../../common/containers/sourcerer';
+import { useGlobalTime } from '../../../../common/containers/use_global_time';
+import { useSearchStrategy } from '../../../../common/containers/use_search_strategy';
+import { useObservedUserDetails } from '../../../../explore/users/containers/users/observed_details';
+import { useFirstLastSeen } from '../../../../common/containers/use_first_last_seen';
+import { useInstalledIntegrations } from '../../../../detections/components/rules/related_integrations/use_installed_integrations';
+import { MANAGED_USER_INDEX, MANAGED_USER_PACKAGE_NAME, MANAGED_USER_QUERY_ID } from './constants';
+import { useQueryInspector } from '../../../../common/components/page/manage_query';
+
+export const useObservedUserItems = (userData: ObservedUserData): ObservedUserTable[] =>
+  useMemo(
+    () =>
+      !userData.details
+        ? []
+        : [
+            { label: i18n.USER_ID, values: userData.details.user?.id, field: 'user.id' },
+            { label: 'Domain', values: userData.details.user?.domain, field: 'user.domain' },
+            {
+              label: i18n.MAX_ANOMALY_SCORE_BY_JOB,
+              field: 'anomalies',
+              values: userData.anomalies,
+            },
+            {
+              label: i18n.FIRST_SEEN,
+              values: userData.firstSeen.date ? [userData.firstSeen.date] : undefined,
+              field: '@timestamp',
+            },
+            {
+              label: i18n.LAST_SEEN,
+              values: userData.lastSeen.date ? [userData.lastSeen.date] : undefined,
+              field: '@timestamp',
+            },
+            {
+              label: i18n.OPERATING_SYSTEM_TITLE,
+              values: userData.details.host?.os?.name,
+              field: 'host.os.name',
+            },
+            {
+              label: i18n.FAMILY,
+              values: userData.details.host?.os?.family,
+              field: 'host.os.family',
+            },
+            { label: i18n.IP_ADDRESSES, values: userData.details.host?.ip, field: 'host.ip' },
+          ],
```

Comment:
`nit`: Might make sense to extract it out to a separate config.

Reply:
yes something like that. The reason I said it because labels and fields are hard coded and the array is quite big for both `userData` and `managedUserDetails`.  You will have to pass `userData`/ `managedUserDetails` to get the complete object.

I don't have strong opinion about it.. you can decide. But at a later stage if more fields are added, I would prefer that [/plugins/security_solution/public/timelines/components/side_panel/new_user_detail/hooks.ts](https://github.com/elastic/kibana/pull/154310/files#diff-06753054db1e98bf65a74aca1ac7a95ec30886ac3ff5bf7411fbcff4857af47d) is not changed at all.